source/common/router/config_impl.h

```diff
@@ -959,6 +959,14 @@ class RouteEntryImplBase : public RouteEntryAndRoute,
   };
 
   using WeightedClusterEntrySharedPtr = std::shared_ptr<WeightedClusterEntry>;
+  // Container for route config elements that pertain to a weightless clusters.
+  // We keep them in a separate data structure to avoid memory overhead for the routes that do not
+  // use weighless clusters.
+  struct WeightedClustersConfig {
+    std::vector<WeightedClusterEntrySharedPtr> weighted_clusters_;
+    uint64_t total_cluster_weight_;
+    std::string random_value_header_name_;
```

Comment:
nit: keep const (possibly by introducing a c'tor)

Reply:
On a second look, why not make all const, by doing local var computations and moving them at the end to a new std::unique_ptr<WeightedClustersConfig>?

Reply:
Done